mixins/provider/test/provider-mixin.test.js

```diff
@@ -0,0 +1,54 @@
+import { provideInstance, ProviderDelegate, requestInstance } from '../provider-mixin.js';
+import { expect } from '@brightspace-ui/testing';
+
+describe('provider-helpers', () => {
+
+	it('should provide number', () => {
+		provideInstance(document, 'instance-number', 42);
+		expect(requestInstance(document.body, 'instance-number')).to.equal(42);
+	});
+
+	it('should provide string', () => {
+		provideInstance(document, 'instance-number', 'forty-two');
+		expect(requestInstance(document.body, 'instance-number')).to.equal('forty-two');
+	});
+
+	it('should provide boolean', () => {
+		provideInstance(document, 'instance-boolean', true);
+		expect(requestInstance(document.body, 'instance-boolean')).to.equal(true);
+	});
+
+	it('should provide function', () => {
+		const someFunction = () => {};
+		provideInstance(document, 'instance-function', someFunction);
+		expect(requestInstance(document.body, 'instance-function')).to.equal(someFunction);
+	});
+
+	it('should provide object', () => {
+		class Beverage {
+			constructor(type) {
+				this.type = type;
+			}
+		}
+		const instance = new Beverage('beer');
+		provideInstance(document, 'instance-object', instance);
+		expect(requestInstance(document.body, 'instance-object')).to.equal(instance);
+	});
+
+	it('should provide delegate result', () => {
+		provideInstance(document, 'instance-delegate', new ProviderDelegate(() => {
+			return 42;
+		}));
+		expect(requestInstance(document.body, 'instance-delegate')).to.equal(42);
+	});
+
+	it('should provide async delegate result', async() => {
+		provideInstance(document, 'instance-async-delegate', new ProviderDelegate(() => {
+			return new Promise(resolve => {
+				setTimeout(() => resolve(42), 200);
+			});
+		}));
```

Comment:
To avoid this test taking > 200ms, you could use Sinon's `useFakeTimers` to advance the clock.

Reply:
Cool yeah, I removed the `setTimeout` altogether. I think simply returning a `Promise` is sufficient to test this cas.